public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -76,7 +76,12 @@ export function PrometheusDatasource(instanceSettings, $q, backendSrv, templateS
       activeTargets.push(target);
 
       var query: any = {};
-      query.expr = templateSrv.replace(target.expr, options.scopedVars, self.interpolateQueryExpr);
+      if (target.expr.indexOf("!==") !== -1 ||  target.expr.indexOf("=~") !== -1 || target.expr.indexOf("!~") !== -1) {
```

Comment:
i see that the templateSrv.replace does a replace of all occurrences of $values. Hence i applied just one check across. it would fail some edge conditions i suppose